### ISSUE TITLE
Slow query execution (/validation?)

### ISSUE THREAD

**Issue by sjchmiela** [rmosolgo/graphql-ruby#720]
For a following query:

```graphql
mutation CreateCalendarTaskMutation(
  $input_0:CreateCalendarTaskInput!,
  $from_1:Date!,
  $from_2:Date!,
  $to_3:Date!,
  $from_4:Date!,
  $from_5:Date!,
  $from_6:Date!,
  $from_7:Date!,
  $to_8:Date!,
  $from_9:Date!,
  $from_a:Date!,
  $from_b:Date!,
  $from_c:Date!
) {
  CreateCalendarTask(input:$input_0) {
    clientMutationId,
    ...F8,
    ...F9,
    ...Fa
  }
}

fragment F0 on User {
  id
}

fragment F1 on Calendar_Support {
  coach_note,
  id,
  coach {
    id
  },
  current_service_obligation {
    id
  }
} 
fragment F2 on Calendar_Task {
  id,
  name,
  completed,
  completed_at,
  description,
  support {
    coach {
      full_name,
      _thumb1eqKWF:thumb(size:24),
      id
    },
  id
  }
} 
fragment F3 on Calendar_Note {
  id,
  content,
  support {
    coach {
      full_name,
      _thumb1eqKWF:thumb(size:24),
      id
    },
  id
  }
} 
fragment F4 on Node {
  id,
  __typename
} 
fragment F5 on Calendar_Task {
  date,
  id,
  name,
  completed
} 
fragment F6 on Calendar_Note {
  id,
  date,
  content
} 
fragment F7 on Calendar_EventsSourceInterface {
  _events4BXBpJ:events(from:$from_1,to:$from_1) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _eventsLsskh:events(from:$from_2,
  to:$to_3) @include(if:true) {
    __typename,
    ...F5,
    ...F6,
    ...F4
  },
  _events1qyPam:events(from:$from_4,
  to:$from_4) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _events18c5Cy:events(from:$from_5,
  to:$from_5) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _events4sc39W:events(from:$from_6,
  to:$from_6) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  __typename,
  _events2X1te2:events(from:$from_7,
  to:$to_8) @include(if:true) {
    __typename,
    ...F5,
    ...F6,
    ...F4
  },
  _events4ku1dP:events(from:$from_9,
  to:$from_9) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _events4svwb3:events(from:$from_a,
  to:$from_a) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _eventsvebYX:events(from:$from_b,
  to:$from_b) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  },
  _events4Dylsf:events(from:$from_c,
  to:$from_c) @include(if:true) {
    __typename,
    ...F2,
    ...F3,
    ...F4
  }
} 
fragment F8 on CreateCalendarTaskPayload {
  service_obligation {
    id
  },
  current_user {
    coach_balance,
    client_balance,
    id,
    ...F0
  },
  support {
    conversation {
      id
    },
    id,
    client {
      full_name,
      _thumb3pGd7o:thumb(size:100),
      id
    },
    __typename,
    coach {
      full_name,
      _thumb1eqKWF:thumb(size:24),
      id
    },
    ...F1,
    ...F7
  }
} 
fragment F9 on CreateCalendarTaskPayload {
  calendar_task_edge {
    node {
      id,
      date,
      support { id }
    },
    cursor
  }
} 
fragment Fa on CreateCalendarTaskPayload {
  support {
    conversation {
      id
    },
    id,
    client {
      full_name,
      _thumb3pGd7o:thumb(size:100),
      id
    },
    __typename,
    coach {
      full_name,
      _thumb1eqKWF:thumb(size:24),
      id
    },
    ...F1,
    ...F7
  }
}
```

time from running `Schema.execute` to running mutation code is about 60 seconds.

What could be the issue? How to debug it?

**Comment by sjchmiela** [comment]
Should this be of any help, after 10 seconds the library is in validation phase:

```
1
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/static_validation/rules/fields_will_merge.rb" line 19 in reduce
2
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/static_validation/rules/fields_will_merge.rb" line 19 in block (2 levels) in validate
3
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/static_validation/rules/fields_will_merge.rb" line 18 in map
4
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/static_validation/rules/fields_will_merge.rb" line 18 in block in validate
5
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 15 in block (5 levels) in visit_each_node
6
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 14 in each
7
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 14 in block (4 levels) in visit_each_node
8
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 26 in each_node
9
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 31 in block (2 levels) in each_node
10
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in each
11
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in block in each_node
12
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each
13
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each_node
14
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 31 in block (2 levels) in each_node
15
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in each
16
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in block in each_node
17
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each
18
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each_node
19
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 31 in block (2 levels) in each_node
20
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in each
21
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 30 in block in each_node
22
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each
23
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 29 in each_node
24
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 13 in block (3 levels) in visit_each_node
25
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 12 in each
26
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 12 in block (2 levels) in visit_each_node
27
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 11 in each
28
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 11 in block in visit_each_node
29
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 9 in each
30
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/internal_representation/visit.rb" line 9 in visit_each_node
31
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/static_validation/validator.rb" line 37 in validate
32
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/query/validation_pipeline.rb" line 74 in ensure_has_validated
33
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/query/validation_pipeline.rb" line 33 in valid?
34
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/query.rb" line 118 in result
35
File "/opt/apps/healthnation_staging/shared/bundle/ruby/2.4.0/gems/graphql-1.5.10/lib/graphql/schema.rb" line 218 in execute
```

**Comment by rmosolgo** [comment]
I've just pushed a branch called `unique-merge`, could you give it a try? 

```ruby
gem "graphql", github: "rmosolgo/graphql-ruby", branch: "unique-merge"
```

**Comment by sjchmiela** [comment]
Yup, that fixes it.

**Comment by rmosolgo** [comment]
Fixed in 1.5.12, thanks for reporting this!
